### PR TITLE
Client Fingerprint Popover Behavior

### PR DESCRIPTION
**Pull request by mylesw27**
This PR intends to find a solution to an added feature request in issue #8924. Currently, the popover that contains the client fingerprint data in the assessment log, disappears without giving an instructor a chance to copy the information. This is due to the current behavior of the Bootstrap popover. 

Bootstrap has essentially three choices when assigning popover behavior. These are "Focus", "Click", and "Hover". "Hover" only displays the popover while the cursor is hovering over the button. "Focus" shows the popover after clicking the button but dismisses the popover as soon as a second click occurs no matter where the click happens. "Click" toggles the popover only when the button is clicked. 

Between these options, the behavior of the popover is not conducive to allowing users to interact with the body of the popover. That is, "Focus" will disappear on any click inside of it and "Click" allows for pesky, lingering popovers (i.e. bad UX). 

This PR intends to add a custom script for allowing the user to click inside the popover to copy or interact with data. However, the script also allows the user to dismiss the popover by clicking anywhere outside of the popover. This will solve the inability to copy useful information or follow links that are contained in the popover body. 

This PR used a similar script in PrairieTest as the basis while stripping it down to use the essentials that were needed here. While doing so, I may have missed some key accessibility features regarding tabbing and using the keyboard to interact with the popover. If you believe there is important functionality that I missed here, please let me know and we can get that added in.

### CODE REVIEW

**Comment by mylesw27** [comment]
This PR will have merge confilicts with #9277 and we should wait until after that PR has been merged before proceeding with this one.

File: package.json
Comment (mylesw27) [inline package.json:49]: It seems as though I incorrectly installed this package. I installed using `yarn add delegated-events` . Is there a preferred way of installing this?

Comment (nwalters512) [reply package.json:49]: If you want to install something just for the `prairielearn` app, you can do either of the following from the root of the repository:

- `yarn workspace @prairielearn/prairielearn add delegated-events`
- `cd apps/prairielearn && yarn add delegated-events`

Comment (mylesw27) [reply package.json:49]: Thanks!

File: apps/prairielearn/assets/scripts/popover.ts
Comment (nwalters512) [inline apps/prairielearn/assets/scripts/popover.ts:1]: I get more confused by this script as I look at it 😅 I also don't think it's actually doing what we want it to. While it works on the fingerprint popovers, it doesn't work on the point/score edit popovers farther up the page.

I'll leave a few more comments inline.

File: apps/prairielearn/assets/scripts/popover.ts
Comment (nwalters512) [inline apps/prairielearn/assets/scripts/popover.ts:38]: This should all be functionality that Bootstrap provides for us; is there a reason this is necessary?

File: apps/prairielearn/assets/scripts/popover.ts
Comment (nwalters512) [inline apps/prairielearn/assets/scripts/popover.ts:45]: This doesn't handle popover content nested arbitrarily deep in the popover body. I'd recommend doing something more like this:

https://github.com/PrairieLearnInc/PrairieTest/blob/3af09500e083e74545d9d3a98c89a800431bf0ec/assets/scripts/behaviors/popover.ts#L121

File: apps/prairielearn/assets/scripts/popover.ts
Comment (nwalters512) [inline apps/prairielearn/assets/scripts/popover.ts:53]: Can you explain what's going on here?

File: package.json
Comment (mylesw27) [reply package.json]: This should be resolved now in [79c8d21](https://github.com/PrairieLearn/PrairieLearn/pull/9286/commits/79c8d211ab90d919e9f5ad7e7c810370f8422d01).

File: apps/prairielearn/assets/scripts/popover.ts
Comment (mylesw27) [reply apps/prairielearn/assets/scripts/popover.ts]: Updated locally in [dbe2bd5](https://github.com/PrairieLearn/PrairieLearn/pull/9286/commits/dbe2bd5b020a6bde994479c70d639a56efa0191e).

File: apps/prairielearn/assets/scripts/popover.ts
Comment (mylesw27) [reply apps/prairielearn/assets/scripts/popover.ts:1]: That's fair, I think I even confused _myself_ on this one. I made a few updates including making it compatible with the other popovers on the page. It is functional but I'm not sure if I am still heading in the right direction here. If you think I am getting further from what needs to happen, maybe we can talk through the code sometime or pair up on it.

File: apps/prairielearn/assets/scripts/popover.ts
Comment (mylesw27) [reply apps/prairielearn/assets/scripts/popover.ts]: I believe this functionality is only provided if you use one of the built-in `data-triggers` for the popover(e.g., `click` or `focus`). We have to use a `manual` data trigger to control the popover which cannot be combined with other triggers according to the Bootstrap docs.

File: apps/prairielearn/assets/scripts/popover.ts
Comment (mylesw27) [reply apps/prairielearn/assets/scripts/popover.ts]: First, we need to close the `openPopover` but also want to mark the popover that we are opening with the current click as the `openingPopover` so that we do no inadvertently close the one we are attempting to open. Then, here we are moving the `openingPopover` to the `openPopover` so that we can close it on the next click and clearing it from `openingPopover` which keeps the popover open. 

The logic here intends to be similar to the `openingPopovers` and `openPopovers` used in the PT popover behavior. However, here we should only ever have one popover open at a time.

File: apps/prairielearn/src/pages/partials/editQuestionPointsForm.ejs
Comment (mylesw27) [reply apps/prairielearn/src/pages/partials/editQuestionPointsForm.ejs]: Removed in [8d0eca5](https://github.com/PrairieLearn/PrairieLearn/pull/9286/commits/8d0eca5ff1dc1248e0ddf281a91b7e52e6cb0d18).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts]: This is a remnant from the original attempt at the code. I thought we needed this as something similar was being used in the PT code and I was trying to get it working here. Now removed in [8d0eca5](https://github.com/PrairieLearn/PrairieLearn/pull/9286/commits/8d0eca5ff1dc1248e0ddf281a91b7e52e6cb0d18).

File: apps/prairielearn/assets/scripts/popover.ts
Comment (mylesw27) [reply apps/prairielearn/assets/scripts/popover.ts]: Updated locally in [8d0eca5](https://github.com/PrairieLearn/PrairieLearn/pull/9286/commits/8d0eca5ff1dc1248e0ddf281a91b7e52e6cb0d18).

File: apps/prairielearn/assets/scripts/popover.ts
Comment (nwalters512) [reply apps/prairielearn/assets/scripts/popover.ts]: This was ultimately resolved with the changes from b1874f5.

File: apps/prairielearn/assets/scripts/popover.ts
Comment (nwalters512) [reply apps/prairielearn/assets/scripts/popover.ts]: This was all changed in b1874f5.

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**
